### PR TITLE
[SM-568] Delete service accounts

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Implement delete service accounts.

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
A couple of minor changes and one question.

File: bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/ServiceAccountRepository.cs [hunk 1]
```diff
@@ -71,6 +81,23 @@ public async Task<IEnumerable<Core.SecretsManager.Entities.ServiceAccount>> GetM
...
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+
+        // Policies can't have a cascade delete, so we need to delete them manually.
+        var policies = dbContext.AccessPolicies.Where(ap =>
+            ids.Contains(((ServiceAccountProjectAccessPolicy)ap).ServiceAccountId!.Value) ||
+            ids.Contains(((GroupServiceAccountAccessPolicy)ap).GrantedServiceAccountId!.Value) ||
+            ids.Contains(((UserServiceAccountAccessPolicy)ap).GrantedServiceAccountId!.Value));
+
```
[the remark this answers is not in the document]
Comment (Hinton) [reply bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/ServiceAccountRepository.cs]: Good catch.

**Review by Thomas-Avery — DISMISSED** [review]
LGTM

**Comment by Hinton** [comment]
I accidentally deleted my gpg public key while reinstalling my machine. Rebased with a signed commit.

**Review by Thomas-Avery — APPROVED** [review]

**Pull request merged**